File: app/src/main/java/tv/hdonlinetv/besttvchannels/movies/watchfree/fragment/LoadingDialogFragment.java
package tv.hdonlinetv.besttvchannels.movies.watchfree.fragment;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import tv.hdonlinetv.besttvchannels.movies.watchfree.databinding.FragmentLoadingDialogBinding;

public class LoadingDialogFragment extends DialogFragment {

    private FragmentLoadingDialogBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentLoadingDialogBinding.inflate(inflater, container, false);
        View view = binding.getRoot();
        binding.loadingAnimation.setAnimation("loading_anim.json");
        binding.loadingAnimation.playAnimation();
        if (getDialog() != null && getDialog().getWindow() != null) {
            getDialog().getWindow().setBackgroundDrawableResource(android.R.color.transparent);
        }
        return view;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/tv/hdonlinetv/besttvchannels/movies/watchfree/fragment/LoadingDialogFragment.kt
package tv.hdonlinetv.besttvchannels.movies.watchfree.fragment

import android.R
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import tv.hdonlinetv.besttvchannels.movies.watchfree.databinding.FragmentLoadingDialogBinding


class LoadingDialogFragment : DialogFragment() {
    private var binding: FragmentLoadingDialogBinding? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentLoadingDialogBinding.inflate(inflater, container, false)
        val view: View = binding!!.getRoot()
        binding!!.loadingAnimation.setAnimation("loading_anim.json")
        binding!!.loadingAnimation.playAnimation()
        if (dialog != null && dialog!!.window != null) {
            dialog!!.window!!.setBackgroundDrawableResource(R.color.transparent)
        }
        return view
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }
}
